internal/cmd/local/command_kafka_start.go

```diff
@@ -0,0 +1,168 @@
+	// 	return err
+	// }
+	// defer out.Close()
+	// io.Copy(os.Stdout, out)
+	// log.CliLogger.Tracef("Pull confluent-local image success")
+
+	err = c.prepareValidPorts()
+	if err != nil {
+		return err
+	}
+
+	if c.Config.LocalPorts == nil {
+		return errors.NewErrorWithSuggestions(errors.FailedToReadPortsErrorMsg, errors.FailedToReadPortsSuggestions)
+	}
+	ports := c.Config.LocalPorts
+	platform := &specsv1.Platform{OS: "linux", Architecture: runtime.GOARCH}
+	config := &container.Config{
+		Image:        imageName,
+		Hostname:     "broker",
+		Cmd:          strslice.StrSlice{"bash", "-c", "'/etc/confluent/docker/run'"},
+		ExposedPorts: nat.PortSet{nat.Port(ports.RestPort + "/tcp"): struct{}{}, nat.Port(ports.PlaintextPort + "/tcp"): struct{}{}},
+		Env: []string{
+			"KAFKA_BROKER_ID=1",
+			"KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT",
+			fmt.Sprintf("KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://broker:%s,PLAINTEXT_HOST://localhost:%s", ports.BrokerPort, ports.PlaintextPort),
+			"KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1",
+			"KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0",
+			"KAFKA_TRANSACTION_STATE_LOG_MIN_ISR=1",
+			"KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR=1",
+			"KAFKA_PROCESS_ROLES=broker,controller",
+			"KAFKA_NODE_ID=1",
+			fmt.Sprintf("KAFKA_CONTROLLER_QUORUM_VOTERS=1@broker:%s", ports.ControllerPort),
+			fmt.Sprintf("KAFKA_LISTENERS=PLAINTEXT://broker:%s,CONTROLLER://broker:%s,PLAINTEXT_HOST://0.0.0.0:%s", ports.BrokerPort, ports.ControllerPort, ports.PlaintextPort),
+			"KAFKA_INTER_BROKER_LISTENER_NAME=PLAINTEXT",
+			"KAFKA_CONTROLLER_LISTENER_NAMES=CONTROLLER",
+			"KAFKA_LOG_DIRS=/tmp/kraft-combined-logs",
+			"KAFKA_REST_HOST_NAME=rest-proxy",
+			fmt.Sprintf("KAFKA_REST_BOOTSTRAP_SERVERS=broker:%s", ports.BrokerPort),
+			fmt.Sprintf("KAFKA_REST_LISTENERS=http://0.0.0.0:%s", ports.RestPort),
+		},
+	}
+	hostConfig := &container.HostConfig{
+		PortBindings: nat.PortMap{
+			nat.Port(ports.RestPort + "/tcp"): []nat.PortBinding{
+				{
+					HostIP:   "0.0.0.0",
+					HostPort: ports.RestPort,
+				},
+			},
+			nat.Port(ports.PlaintextPort + "/tcp"): []nat.PortBinding{
+				{
+					HostIP:   "0.0.0.0",
+					HostPort: ports.PlaintextPort,
+				},
+			},
+		},
+	}
+
+	createResp, err := dockerClient.ContainerCreate(context.Background(), config, hostConfig, nil, platform, "confluent-local")
+	if err != nil {
+		return errors.CatchContainerNameInUseError(err)
+	}
+	log.CliLogger.Tracef("Create confluent-local container success")
+
+	err = dockerClient.ContainerStart(context.Background(), createResp.ID, types.ContainerStartOptions{})
+	if err != nil {
+		return err
+	}
+	log.CliLogger.Tracef("Start confluent-local container success")
+
+	output.Printf("Started Confluent local container %v.\nContinue your experience with Confluent local running `confluent local kafka produce` and `confluent local kafka consume`.\n", createResp.ID[:10])
+	return nil
+}
```

Comment:
This function is about 86 lines long. Can we make it shorter? The configuration seems like an easy thing to break out and unit test separately.